src/render/painter.js

```diff
@@ -837,11 +837,15 @@ class Painter {
         if (fog) {
             const temporalOffset = (this.frameCounter / 1000.0) % 1;
             const fogColor = fog.properties.get('color');
+            const hazeColor = fog.properties.get('haze-color');
             const uniforms = {};
 
             uniforms['u_cam_matrix'] = tileID ? this.transform.calculateCameraMatrix(tileID) : this.identityMat;
             uniforms['u_fog_range'] = fog.properties.get('range');
             uniforms['u_fog_color'] = [fogColor.r, fogColor.g, fogColor.b];
+            uniforms['u_haze_color_linear'] = [hazeColor.r, hazeColor.g, hazeColor.b].map(c => 1.5 * Math.pow(c, 2.2));
```

Comment:
I don't like this factor of 1.5 (totally arbitrary, chosen to put colors in a reasonable magnitude), and would not be against just making this the mentioned additional style property (`haze-magnitude`, `haze-amount`, `haze-intensity` or something)

Reply:
Oh, there's alpha as a way to specify the amount of haze, I suppose. The only downside there is that you're still clipped to kind of an arbitrary range since alpha = [0, 1]. What is "normal" haze? alpha = 0.2??